### PR TITLE
improved authentication logging

### PR DESCRIPTION
**Pull request by chaudum**
* log failed attempts at level WARN
* removed unused return values
* make method package-private

### CODE REVIEW

File: sql/src/main/java/io/crate/protocols/postgres/Messages.java [hunk 1]
```diff
@@ -189,12 +189,15 @@ public void operationComplete(ChannelFuture future) throws Exception {
...
 
-    public static ChannelFuture sendAuthenticationError(Channel channel, String message) {
+    static void sendAuthenticationError(Channel channel, String message) {
+        if (LOGGER.isWarnEnabled()) {
+            LOGGER.warn(message);
+        }
         byte[] msg = message.getBytes(StandardCharsets.UTF_8);
         byte[] severity = "FATAL".getBytes(StandardCharsets.UTF_8);
         byte[] errorCode = "28000".getBytes(StandardCharsets.UTF_8);
         byte[] method = "ClientAuthentication".getBytes(StandardCharsets.UTF_8);
```
Comment (dobe) [inline sql/src/main/java/io/crate/protocols/postgres/Messages.java]: can we make those byte[] vars static fields? and even remove the method args?

File: users/src/main/java/io/crate/http/netty/HttpAuthUpstreamHandler.java [hunk 2]
```diff
@@ -98,6 +97,9 @@ private void handleHttpChunk(ChannelHandlerContext ctx, MessageEvent e) {
 
     @VisibleForTesting
     static void sendUnauthorized(Channel channel, @Nullable String body) {
+        if (LOGGER.isWarnEnabled() && body != null) {
```
Comment (dobe) [inline users/src/main/java/io/crate/http/netty/HttpAuthUpstreamHandler.java]: warn condition also not needed
what does this mean when body is null? might be interesting too?

Comment (chaudum) [inline users/src/main/java/io/crate/http/netty/HttpAuthUpstreamHandler.java]: body is `null` if the handler receives a HttpChunk but is not authorized. IMO it is ok to not send an additional message in the HTTP response body

Comment (chaudum) [inline users/src/main/java/io/crate/http/netty/HttpAuthUpstreamHandler.java]: honestly I don't know when this edge case is going to happen

**Pull request merged**